tests/integration/accounts/test_contract_account.py

```diff
@@ -91,3 +94,19 @@ async def test_should_give_infinite_allowance_to_kakarot(
                 )
                 == "Uint256(low=340282366920938463463374607431768211455, high=340282366920938463463374607431768211455)"
             )
+
+        async def test_should_take_deployment_fees(self, kakarot, eth):
+            evm_address = int(generate_random_evm_address(), 16)
+            computed_starknet_address = (
+                await kakarot.compute_starknet_address(evm_address).call()
+            ).result[0]
+
+            amount = 100000
+            await fund_evm_address(evm_address, kakarot, eth, amount)
+
+            await kakarot.deploy_externally_owned_account(evm_address).execute()
```

Comment:
could we use a non trivial `caller_address` so as to make sure that funds are not just sent to address 0

Reply:
It makes no difference from what I could see, the caller address is 0 regardless of whatever caller address you choose.

Reply:
there is plenty of example, even in this file, where we use `.execute(caller_address=xxx)` that wouldn't pass otherwise so I think it's working. can you double check?

Reply:
Yeah, I have tried it, the reason how I explored this issue was when I started seeing the error, cannot transfer to 0 address from the ERC20 library!

I tried changing the value multiple times, but it looked like the caller address from tx.info was always 0! But, I am fixing few tests, and lemme recheck again and get back!